go-controller/pkg/ovn/base_network_controller_policy.go

```diff
@@ -519,106 +501,124 @@ func getPolicyType(policy *knet.NetworkPolicy) (bool, bool) {
-		// 2. the gotten LSP is scheduled for removal (stateful-sets).
-		portInfo, err := oc.logicalPortCache.get(pod, types.DefaultNetworkName)
-		if err != nil {
-			klog.Warningf("Failed to get LSP for pod %s/%s for networkPolicy %s, err: %v",
-				pod.Namespace, pod.Name, np.name, err)
-			errObjs = append(errObjs, pod)
-			continue
-		}
+		nadNames := bnc.getPodNADNames(pod)
+		for _, nadName := range nadNames {
+			var logicalPortName string
+			if !bnc.IsSecondary() {
+				logicalPortName = util.GetLogicalPortName(pod.Namespace, pod.Name)
+			} else {
+				logicalPortName = util.GetSecondaryNetworkLogicalPortName(pod.Namespace, pod.Name, nadName)
+			}
 
-		// Add pod to errObjs if LSP is scheduled for deletion
-		if !portInfo.expires.IsZero() {
-			klog.Warningf("Stale LSP %s for network policy %s found in cache",
-				portInfo.name, np.name)
-			errObjs = append(errObjs, pod)
-			continue
-		}
+			if _, ok := np.localPods.Load(logicalPortName); ok {
+				// port is already added for this policy
+				continue
+			}
+
+			// Return error for retry if
+			// 1. getting pod LSP from the cache fails,
+			// 2. the gotten LSP is scheduled for removal (stateful-sets).
+			portInfo, err := bnc.logicalPortCache.get(pod, nadName)
+			if err != nil {
+				klog.Warningf("Failed to get get LSP for pod %s/%s NAD %s for networkPolicy %s, err: %v",
+					pod.Namespace, pod.Name, nadName, np.name, err)
+				errs = append(errs, fmt.Errorf("unable to get port info for pod %s/%s NAD %s", pod.Namespace, pod.Name, nadName))
+				continue
+			}
 
-		// LSP get succeeded and LSP is up to fresh
-		klog.V(5).Infof("Fresh LSP %s for network policy %s found in cache",
-			portInfo.name, np.name)
+			// Add pod to errObjs if LSP is scheduled for deletion
+			if !portInfo.expires.IsZero() {
+				klog.Warningf("Stale LSP %s for network policy %s found in cache",
+					portInfo.name, np.name)
+				errs = append(errs, fmt.Errorf("unable to get port info for pod %s/%s NAD %s", pod.Namespace, pod.Name, nadName))
+				continue
+			}
+
+			// LSP get succeeded and LSP is up to fresh
+			klog.V(5).Infof("Fresh LSP %s for network policy %s found in cache",
+				portInfo.name, np.name)
 
-		policyPortUUIDs = append(policyPortUUIDs, portInfo.uuid)
-		policyPortsToUUIDs[portInfo.name] = portInfo.uuid
+			policyPortUUIDs = append(policyPortUUIDs, portInfo.uuid)
+			policyPortsToUUIDs[portInfo.name] = portInfo.uuid
+		}
 	}
+
 	return
 }
 
 // getExistingLocalPolicyPorts will find and return port info for every given pod obj, that is present in np.localPods.
-// if there are problems with fetching port info from logicalPortCache, pod will be added to errObjs.
-func (oc *DefaultNetworkController) getExistingLocalPolicyPorts(np *networkPolicy,
-	objs ...interface{}) (policyPortsToUUIDs map[string]string, policyPortUUIDs []string, errObjs []interface{}) {
+// if there are problems with fetching port info from logicalPortCache, error will be added to return error array.
```

Comment:
looks like errors are not returned anymore, probably we can delete that comment line